modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/DefaultRouteHandling.scala

```diff
@@ -19,7 +27,10 @@ trait DefaultRouteHandling {
   def combinedRoutesFor(initialPrefix: String): Route = handleExceptions(ExceptionHandling.exceptionHandler) {
     handleRejections(RejectionHandling.rejectionHandler) {
       pathPrefix(initialPrefix) {
-        resourceRoutes ~ searchRoutes
+        extractCredentials {
+          case Some(cred: OAuth2BearerToken) => resourceRoutes(cred) ~ searchRoutes(cred)
+          case _                             => complete(StatusCodes.Unauthorized)
```

Comment:
Anonymous should be allowed if permissions allow; lack of token -> anonymous.

Maybe we should abstract this under a caller type:
```
sealed trait Caller extends Product with Serializable {
  def identities: Set[Identity]
  def accessToken: Option[String]
}
object Caller {
  final case object AnonymousCaller extends Caller {
    override val identities = Set(Anonymous)
    override val accessToken = None
  }
  final case class AuthenticatedCaller(token: String, info: UserInfo) extends Caller {
    override val identities: Set[Identity] = ??? // compute from info
    override val accessToken = Some(token)
  }
}
```
... and have the authentication directive yield a `Caller` user by using an AsyncAuthenticator[Caller].
Thoughts?